code/modules/projectiles/guns/energy/special.dm

```diff
@@ -364,3 +364,128 @@
+/obj/item/projectile/bullet/reusable/railgun_rod
+	name = "flying rod"
+	desc = "It's heading fast towards your face!"
+	icon_state = "chronobolt"	//that is, till sprites make one better
+	ammo_type = /obj/item/stack/rods
+	damage_type = BRUTE
+	damage = 35
+	var/embed_chance = 75
+	var/obj/item/stack/rods/rod_drop
+	
+/obj/item/projectile/bullet/reusable/railgun_rod/on_hit(atom/target, blocked)
+	. = ..()
+	var/mob/living/carbon/human/human_target = target
+	if(istype(human_target))
+		if (prob(embed_chance))
+			var/obj/item/stack/rods/embed_rod = new ammo_type(human_target)
+			embed_rod.add(3 - 1)
+			embed_rod.update_icon()
+			embed_rod.embedding = new /datum/embedding_behavior(
+				embed_chance = 0,
+                embedded_fall_chance = 20,
+                embedded_ignore_throwspeed_threshold = TRUE
+			)
+			human_target.embed_object(embed_rod)
+			if (dropped && rod_drop)
+				qdel(rod_drop)
+
+/obj/item/projectile/bullet/reusable/railgun_rod/handle_drop()
+	if(!dropped)
+		var/turf/T = get_turf(src)
+		rod_drop = new ammo_type(T)
+		rod_drop.add(3-1)
+		dropped = TRUE
+
+/obj/item/projectile/bullet/reusable/railgun_rod/on_hit(atom/target, blocked)
+	if(..())	
+		var/mob/living/ltarget = target
+		if (istype(ltarget))
+			var/atom/throw_target = get_edge_target_turf(ltarget, angle2dir(Angle))
+			ltarget.throw_at(throw_target, 4, 2)
+		
+// --- THE GUN ---
+
+/obj/item/gun/energy/railgun
+	icon_state = "export_cannon"
+	name = "rail gun"
+	desc = "A heavy gun that uses magnetic induction to launch iron rods at high velocity."
+	ammo_type = list(/obj/item/ammo_casing/energy/railgun_rod/strong)
+	dead_cell = FALSE
+	can_suppress = FALSE
+	can_bayonet = FALSE
+	can_flashlight = FALSE
+	var/rod_loaded = TRUE
+
+/obj/item/gun/energy/railgun/update_icon(force_update)
+	if (!cell && cell.charge<1000)
+		icon_state = "export_cannon_empty"
+	else 
+		icon_state = "export_cannon"
+
+/obj/item/gun/energy/railgun/attack_self(mob/living/user as mob)
+	if(rod_loaded)
+		var/obj/item/stack/rods/mylongrod = new /obj/item/stack/rods(get_turf(src))
+		mylongrod.add( 3 - 1 )
+		rod_loaded = FALSE
+		to_chat(user, "<span class='warning'>You unload [mylongrod.amount] iron rods from the [src]!</span>")
+
+/obj/item/gun/energy/railgun/attackby(obj/item/item, mob/user, params)
+	if (istype(item,/obj/item/stack/rods) && do_after(user,60,target = src))
+		playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You load [src] with the [item]!</span>")
+		var/obj/item/stack/rods/payload = item
+		var/load_amount = 3
+		if (payload.use(load_amount,FALSE,TRUE))
+			payload.update_icon()
+			rod_loaded += TRUE
+			update_icon(FALSE)		
+	
+/obj/item/gun/energy/railgun/recharge_newshot(no_cyborg_drain)
+	if (!rod_loaded)
+		return FALSE
+	. = ..()
+
+/obj/item/gun/energy/railgun/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	. = ..()
+	if (.)
+		rod_loaded=false
```

Comment:
bools have to be fully capitalized, and it's good practice to put a space between the `=`